### PR TITLE
Enhancement/list load on demand

### LINKED ISSUE #590: [Enhancement] Enable LoadOnDemand after disabling
**Is your enhancement request related to a problem? Please describe.**
If i have a `kirby-list` with the `loadOnDemand` callback set, i can disable `loadOnDemand` by calling  `event.complete(true)` on the `LoadOnDemandEventData` event.

But i can not enable `LoadOnDemand` again.

**Describe the solution you'd like**
I would a simple way to enable the `loadOnDemand` events again.

Right now the `kirby-list` has a property called `isLoadOnDemandEnabled`.
If this was made an `@Input()` you would be able to set as a user of the `kirby-list` and that way enable `loadOnDemand` again.

**Describe alternatives you've considered**
The more pretty solution would be that the `kirby-list` would have a public method to enable/disable `loadOnDemand`.

### PR DESCRIPTION
**Pull request by MadsBuchmann**
This PR closes #590

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) (will be done after draft review)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #590 

## What is the new behavior?
A `enableLoadOnDemand` public method has been added to the list component. If this is called with `true` it will enable the emission of the `loadOnDemand`-event, vice versa for `false`. 

This can be used to re-enable emission of the `loadOnDemand` event after it has been disabled using the `.complete()` function from the `loadOnDemand` event. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Other Information 
I have re-ordered imports in the files i've been working in as the linter threw a warning. 

Also i started out solving the issue using an input property, which is why there's an excessive amount of commits. Eventually refactored to create a public method, as this seemed like the simplest solution to me. This is why it might seem like there's an excessive amount of commits 😊

### CODE REVIEW

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 1]
```diff
@@ -142,16 +143,26 @@ describe('ListComponent', () => {
   describe('function: ngOnInit', () => {
     it('should enable load more, if there is a subscriber to the loadMore event emitter', () => {
       component.loadOnDemand.subscribe((_: LoadOnDemandEvent) => {});
-
       component.ngOnInit();
-
-      expect(component.isLoadOnDemandEnabled).toBeTruthy();
+      expect(component.isLoadOnDemandEnabled).toBeTrue();
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:147]: The line break is intentional to denote a arrange-act-assert pattern:
```suggestion

      expect(component.isLoadOnDemandEnabled).toBeTrue();
```

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 2]
```diff
@@ -142,16 +143,26 @@ describe('ListComponent', () => {
...
 
-      expect(component.isLoadOnDemandEnabled).toBeFalsy();
+      expect(component.isLoadOnDemandEnabled).toBeFalse();
+    });
+  });
+
+  describe('function: enableLoadOnDemand', () => {
+    it('should enable load more, if called with true', () => {
+      component.enableLoadOnDemand(true);
+      expect(component.isLoadOnDemandEnabled).toBeTrue();
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:160]: Add a line break to denote a arrange-act-assert pattern (no arrange in this one):
```suggestion
      component.enableLoadOnDemand(true);
      
      expect(component.isLoadOnDemandEnabled).toBeTrue();
```

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 3]
```diff
@@ -142,16 +143,26 @@ describe('ListComponent', () => {
...
+    });
+
+    it('should disable load more, if called with false', () => {
+      component.enableLoadOnDemand(false);
+      expect(component.isLoadOnDemandEnabled).toBeFalse();
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:165]: Add a line break to denote a arrange-act-assert pattern (no arrange in this one):
```suggestion
      component.enableLoadOnDemand(false);
      
      expect(component.isLoadOnDemandEnabled).toBeFalse();
```

File: libs/designsystem/src/lib/components/list/list.component.ts [hunk 4]
```diff
@@ -150,6 +151,7 @@ export class ListComponent implements OnInit, AfterViewInit, OnChanges {
     this.hasDeprecatedItemTemplate = !!this.legacyItemTemplate || !!this.legacyFlexItemTemplate;
     this.initializeSwipeActions();
     this.isSelectable = this.itemSelect.observers.length > 0;
+
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.ts:154]: Any reason for this line break?

File: libs/designsystem/src/lib/components/list/list.component.ts [hunk 5]
```diff
@@ -238,4 +240,8 @@ export class ListComponent implements OnInit, AfterViewInit, OnChanges {
       this.isSwipingEnabled = this.platform.isTouch();
     }
   }
+
+  public enableLoadOnDemand(isLoadOnDemandEnabled: boolean) {
+    this.isLoadOnDemandEnabled = isLoadOnDemandEnabled;
+  }
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.ts:246]: Couldn't we just make the existing property `isLoadOnDemandEnabled` an `@Input` property?
Why the need for a method?

**Review by jakobe — CHANGES_REQUESTED** [review]
Couldn't we just make the existing property `isLoadOnDemandEnabled` an `@Input` property?
Why the need for a method?

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 1 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts:149]: Should be resolved by f1d733e.

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 2 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts]: Should be resolved by f1d733e.

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 3 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts]: Should be resolved by f1d733e.

File: libs/designsystem/src/lib/components/list/list.component.ts (hunk 4 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.ts]: Not at all 😄 - should be resolved by f311d2d.

File: libs/designsystem/src/lib/components/list/list.component.ts (hunk 5 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.ts:245]: With my current understanding of Angular; making `isLoadOnDemandEnabled` an `@input` property would move the responsibility of determining the state of `isLoadOnDemandEnabled` to the parent component that is utilising the `list` component. But the state can currently be altered by the `done` function exposed by the `loadOnDemand` event. 

Marking `isLoadOnDemandEnabled` with `@input` would result In there being two ways the value can be altered: with the input property and via the `done` function.  

The parent component will not be notified if the state of `isLoadOnDemandEnabled` is changed via `done` (unless an event emitter or callback is implemented). To my understanding, this will result in `isLoadOnDemand` being perceived as having two different values: 
   1. The parent component trying to control `isLoadOnDemandEnabled` via an input property will have the initial value. 
   2. The `list` component with the `isLoadOnDemandEnabled` member will have the altered value that is set by `done`. 

Now that I think about it - my understanding of how internal changes to input properties are handled might be wrong. But even if an internal change to an input property is reflected in the parent, it would mean we're breaking the unidirectional dataflow.

 My tought was that It was simpler to keep the responsibility of controlling the state of `isLoadOnDemand` in the `list`-component, as we're already using `done` to control the state of `isLoadOnDemandEnabled`. The component can then be "told" what we want with `enableLoadOnDemand`, moving the responsibility of setting the value back to the component itself. 

My own preferred solution would be to remove the `done` function and then only being able to control the state of `isLoadOnDemandEnabled` via an input property as this to me is cleaner. But I suppose there's a reason why it is done via an function on an event and altering it might be a breaking change 😄

Comment (jakobe) [reply libs/designsystem/src/lib/components/list/list.component.ts:245]: That might be true under other circumstances, like encapsulating the state management to the component, like having readonly props and some checks when trying to alter state through a method.

But in this particular case, it's still the consumer that sets the state - through a method, but tomato/tomato - and there are no checks/guards, side effects or anything going on... So it's kinda redundant to wrap that property in a method (and by the way: The property is already public, so anyone can still set it ;) )

In an Angular context ( or any declarative context really), by exposing the property it's possible to do this:
`<kirby-list [isLoadOnDemandEnabled]="someAsyncOperationReturningBoolean | async">`

With a method, we would have to expose the list instance through a query and call the method on that instance when the value changes. It's doable, but not as straight forward.

Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.ts:245]: I see your point and will revert to using an input property instead. I've got some follow up questions which I might bother you with Monday 😄

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.spec.ts [hunk 6]
```diff
@@ -69,6 +72,19 @@ describe('InfiniteScrollDirective', () => {
       expect(directive.scrollEnd.emit).toHaveBeenCalledTimes(1);
     }));
 
+    it('should emit event if initalized as disabled and then enabled while more than 80% of the element has been scrolled', fakeAsync(() => {
```
Comment (MadsBuchmann) [inline libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.spec.ts:75]: The wording on this is a bit clumsy - I will change it if I come up with something better. Suggestions would be appreciated tho 😀

**Comment by MadsBuchmann** [comment]
I've updated the infiniteScrollDirective. If it was initialised as disabled, it would not subscribe to the scroll$-observable if it was enabled later. 

This resulted in some unexpected behaviour when initialising a list with `isLoadOnDemandEnabled = false` and then changing it to `true`, as the loadMore event would not be emitted.

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts [hunk 7]
```diff
@@ -24,7 +24,7 @@ export const INFINITE_SCROLL_DEBOUNCE = 100;
 @Directive({
   selector: '[kirbyInfiniteScroll]',
 })
-export class InfiniteScrollDirective implements AfterViewInit, OnDestroy {
+export class InfiniteScrollDirective implements OnDestroy, AfterViewInit {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts:27]: Why is this sorted backwards?

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts [hunk 8]
```diff
@@ -1,16 +1,16 @@
...
-  Output,
-  ElementRef,
   NgZone,
+  OnDestroy,
+  Output,
 } from '@angular/core';
-import { Subject, fromEvent } from 'rxjs';
-import { debounceTime, takeUntil, filter, map } from 'rxjs/operators';
+import { fromEvent, Subject } from 'rxjs';
+import { debounceTime, filter, map, takeUntil, tap } from 'rxjs/operators';
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts:13]: Remove unused import
```suggestion
import { debounceTime, filter, map, takeUntil } from 'rxjs/operators';
```

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts [hunk 9]
```diff
@@ -54,8 +61,30 @@ export class InfiniteScrollDirective implements AfterViewInit, OnDestroy {
...
+
+  /**
+   * trigger {@link ngUnsubscribe} complete on component destroy lifecycle hook
+   */
+  ngOnDestroy(): void {
+    this.ngUnsubscribe$.next();
+    this.ngUnsubscribe$.complete();
+  }
+
+  private checkScroll$Subscription(disabled: boolean) {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts:86]: The `$` suffix is a naming convention to denote observable variables - we don't use it inside member names

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts [hunk 10]
```diff
@@ -34,7 +34,14 @@ export class InfiniteScrollDirective implements AfterViewInit, OnDestroy {
   /**
    * If true then {@link scrollEnd} event should NOT be emitted
    */
-  @Input() disabled = false;
+  _disabled: boolean = false;
+  @Input() set disabled(value: boolean) {
+    this.checkScroll$Subscription(value);
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts:39]: Pretty sure it would suffice just to remove the `disabled` guard at the top of `ngAfterViewInit` as it's already using the `disabled` property as a filter inside the pipe...

File: libs/designsystem/src/lib/components/list/helpers/list-helper.spec.ts [hunk 11]
```diff
@@ -1,9 +1,8 @@
 import { EventEmitter } from '@angular/core';
 
 import { ListComponent } from '../list.component';
-import { LoadOnDemandEvent } from '../list.event';
-
 import { ListHelper } from './list-helper';
+import { LoadOnDemandEvent } from '../list.event';
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/helpers/list-helper.spec.ts]: Is this change related to the PR?

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 12]
```diff
@@ -141,32 +141,60 @@ describe('ListComponent', () => {
   });
 
   describe('function: ngOnInit', () => {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:143]: Be a good boy scout 🤞 
```suggestion
  describe('function: ngAfterViewInit', () => {
```

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 13]
```diff
@@ -141,32 +141,60 @@ describe('ListComponent', () => {
...
+        component.loadOnDemand.subscribe((_: LoadOnDemandEvent) => {});
+      });
 
-      component.ngOnInit();
+      it('should enable load more, if isLoadOnDemandEnabled is not set', () => {
+        component.isLoadOnDemandEnabled = undefined;
 
-      expect(component.isLoadOnDemandEnabled).toBeTrue();
-    });
+        component.ngOnInit();
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:152]: Be a good boy scout 🤞
```suggestion
        component.ngAfterViewInit();
```

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 14]
```diff
@@ -141,32 +141,60 @@ describe('ListComponent', () => {
...
-      expect(component.isLoadOnDemandEnabled).toBeFalse();
-    });
-  });
+      it('should enable load more, if isLoadOnDemandEnabled is set to true', () => {
+        component.isLoadOnDemandEnabled = undefined;
 
-  describe('function: enableLoadOnDemand', () => {
-    it('should enable load more, if called with true', () => {
-      component.enableLoadOnDemand(true);
+        component.ngOnInit();
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:160]: Be a good boy scout 🤞
```suggestion
        component.ngAfterViewInit();
```

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 15]
```diff
@@ -141,32 +141,60 @@ describe('ListComponent', () => {
   });
 
   describe('function: ngOnInit', () => {
-    it('should enable load more, if there is a subscriber to the loadMore event emitter', () => {
-      component.loadOnDemand.subscribe((_: LoadOnDemandEvent) => {});
+    describe('there is a subscriber to the loadMore event', () => {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:144]: Consider wording
```suggestion
    describe('when there is a subscriber to the loadMore event', () => {
```

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 16]
```diff
@@ -141,32 +141,60 @@ describe('ListComponent', () => {
...
+
+        component.ngOnInit();
+
+        expect(component.isLoadOnDemandEnabled).toBeFalse();
+      });
     });
 
-    it('should disable load more, if called with false', () => {
-      component.enableLoadOnDemand(false);
+    describe('there is no subscriber to the loadMore event', () => {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:174]: Consider wording:
```suggestion
    describe('when there is no subscriber to the loadMore event', () => {
```

File: libs/designsystem/src/lib/components/list/list.component.spec.ts [hunk 17]
```diff
@@ -141,32 +141,60 @@ describe('ListComponent', () => {
...
+      it('should disable load more, if isLoadOnDemandEnabled is not set', () => {
+        component.isLoadOnDemandEnabled = undefined;
 
-      expect(component.isLoadOnDemandEnabled).toBeFalse();
+        component.ngOnInit();
+
+        expect(component.isLoadOnDemandEnabled).toBeFalse();
+      });
+
+      it('should enable load more, if isLoadOnDemandEnabled is set to true', () => {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.spec.ts:183]: Should it? Isn't there supposed to be at least 1 subscriber for this to be enabled? 🤔

**Review by jakobe — CHANGES_REQUESTED** [review]
Pretty sure it would suffice just to remove the `disabled` guard at the top of `ngAfterViewInit` as it's already using the `disabled` property as a filter inside the pipe...

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 16 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts]: Should be resolved by bf284f7

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 15 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts]: Should be resolved by bf284f7

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts (hunk 9 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts]: Should be resolved by 6730130

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 16 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts]: Should be resolved by bf284f7

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts (hunk 8 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts]: Should be resolved by 33362be

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts (hunk 7 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts]: Should be resolved by 33362be

File: apps/cookbook/src/app/examples/list/components/load-on-demand/list-load-on-demand-example.component.ts [hunk 18]
```diff
@@ -1,9 +1,8 @@
 import { Component } from '@angular/core';
 import { of } from 'rxjs';
 
-import { LoadOnDemandEvent } from '@kirbydesign/designsystem';
-
 import { BaseListComponent } from '../base-list.component';
+import { LoadOnDemandEvent } from '@kirbydesign/designsystem';
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply apps/cookbook/src/app/examples/list/components/load-on-demand/list-load-on-demand-example.component.ts:3]: Should be resolved by 33362be

File: libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts (hunk 10 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts]: You're right! - should be resolved by: 96fa29c

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 13 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts:152]: I'm not sure I understand this suggestion - the function being tested is ngOnInit 😅

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 14 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts:160]: I'm not sure I understand this suggestion - the function being tested is ngOnInit 😅

File: apps/cookbook/src/app/examples/list/components/load-on-demand/list-load-on-demand-example.component.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply apps/cookbook/src/app/examples/list/components/load-on-demand/list-load-on-demand-example.component.ts:4]: I've reverted the file back to original state from master, as it is no longer related to this pull request.

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html [hunk 19]
```diff
@@ -1,4 +1,29 @@
 <div class="example">
+  <p>
+    The kirby list will emit at <code>loadMore</code> event, when the user has scrolled 80% of the
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html:3]: Typo
```suggestion
    The kirby list will emit a <code>loadMore</code> event, when the user has scrolled 80% of the
```

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html [hunk 20]
```diff
@@ -1,4 +1,29 @@
...
+    list. It is the responsibility of the consumer to add more items to the bound
+    <code>items</code>.
+  </p>
+  <p>
+    The event emitted is of type <code>KirbyLoadMoreEvent</code> which exposes a
+    <code>complete</code> callback function.
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html:9]: Consider readability:
```suggestion
    <code>complete()</code> callback function.
```

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html [hunk 21]
```diff
@@ -1,4 +1,29 @@
...
+  </p>
+  <p>
+    The <code>complete</code> callback function should be called when the loading of the additional
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html:12]: Consider readability:
```suggestion
    The <code>complete()</code> callback function should be called when the loading of the additional
```

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html [hunk 22]
```diff
@@ -1,4 +1,29 @@
...
+    items are done.
+  </p>
+  <p>
+    If <code>complete</code> is called with the <code>disableLoadMore</code> argument set to
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html:16]: Consider readability:
```suggestion
    If <code>complete(disableLoadMore?: boolean)</code> is called with the <code>disableLoadMore</code> argument set to
```

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html [hunk 23]
```diff
@@ -1,4 +1,29 @@
...
+    <code>true</code>, no further events will be emitted. Further more, if any is given, the
+    <code>noMoreItemsText</code> will be shown.
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html:18]: Consider wording:
```suggestion
    <code>true</code>, no further events will be emitted. If <code>noMoreItemsText</code> is configured it will be shown at the end of the list at this point.
```

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html [hunk 24]
```diff
@@ -1,4 +1,29 @@
...
+  </p>
   <cookbook-list-load-on-demand-example></cookbook-list-load-on-demand-example>
   <cookbook-code-viewer [html]="exampleHtml"></cookbook-code-viewer>
+  <h2>Properties:</h2>
+  <cookbook-showcase-properties [properties]="properties"></cookbook-showcase-properties>
+  <h2>Events:</h2>
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html:24]: Consider importing `@import '../showcase.shared';` in `apps/cookbook/src/app/examples/list-load-on-demand-example/list-load-on-demand-example.component.scss` to get default spacing between headings (e.g. see https://cookbook.kirby.design/home/showcase/modal and `apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.scss` for an example)

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.ts [hunk 25]
```diff
@@ -9,4 +12,34 @@ declare var require: any;
...
+    {
+      name: 'isLoadOnDemandEnabled',
+      description: 'Determines if the loadOnDemand event should be emitted.',
+      inputValues: ['boolean'],
+      defaultValue: 'If there is a subscriber to the loadOnDemand event: true, otherwise false.',
+    },
+    {
+      name: 'noMoreItemsText',
+      description:
+        'Text to display when no more items can be loaded (used for "on demand"-loading).',
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.ts:26]: Consider wording
```suggestion
        'Text to display when on demand loading of items is complete.,
```

File: libs/designsystem/src/lib/components/list/helpers/list-helper.ts [hunk 26]
```diff
@@ -1,5 +1,5 @@
-import { ListComponent } from '../list.component';
 import { LoadOnDemandEventData } from '../list.event';
+import { ListComponent } from '../list.component';
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/helpers/list-helper.ts]: Seems like the order has been reversed? Please revert :)

File: libs/designsystem/src/lib/components/list/list.component.ts [hunk 27]
```diff
@@ -104,6 +99,13 @@ export class ListComponent implements OnInit, AfterViewInit, OnChanges {
   @Input()
   hasItemSpacing: boolean;
 
+  /**
+   * Determines if the loadOnDemand event should be emitted
+   * - will be set to true if not given a value by parent component
+   * and there's a subscriber to the loadOnDemand event
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/list/list.component.ts:105]: Consider wording:
```suggestion
   * Determines if the loadOnDemand event should be emitted.
   * Will default to true if there is at least one subscriber to the loadOnDemand event
```

**Review by jakobe — CHANGES_REQUESTED** [review]

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 13 above)
Comment (jakobe) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts:152]: True, that is the function being called in the test, but it's actually being implemented in the `ngAfterViewInit()`function (which automatically gets called after calling `ngOnInit()`)

File: libs/designsystem/src/lib/components/list/helpers/list-helper.ts (hunk 26 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/helpers/list-helper.ts]: It's intentional - I have reverted the file in commit fdaac63 as no other changes than reordering imports are being applied to that file 😄 

It should be identical to `list-helper.ts` on master now.

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 13 above)
Comment (jakobe) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts:152]: My bad 🙈

**Review by jakobe — APPROVED** [review]

File: libs/designsystem/src/lib/components/list/list.component.spec.ts (hunk 17 above)
Comment (MadsBuchmann) [reply libs/designsystem/src/lib/components/list/list.component.spec.ts:183]: As discussed in person I've removed the test case.

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html (hunk 24 above)
Comment (MadsBuchmann) [reply apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html]: Resolved by: 6e9a684

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html [hunk 28]
```diff
@@ -1,4 +1,33 @@
...
+    <code>disableLoadMore</code> argument set to <code>true</code>, no further events will be
+    emitted. If <code>noMoreItemsText</code> is configured it will be shown at the end of the list
+    at this point.
+  </p>
+</div>
+<div class="page-example">
   <cookbook-code-viewer [html]="exampleHtml"></cookbook-code-viewer>
+  <cookbook-iphone src="/examples/list-load-on-demand" showExternalLink="true"></cookbook-iphone>
 </div>
+
```
Comment (MadsBuchmann) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.html:26]:
```suggestion
```

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.scss [hunk 29]
```diff
@@ -1,6 +1,8 @@
 @import '../showcase.shared';
 
-fieldset {
-  display: inline-block;
-  margin-bottom: size('s');
+.page-example {
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.scss:3]: Consider naming (I believe this is coming from `Page` example?)
```suggestion
.example-wrapper {
```

**Review by jakobe — APPROVED** [review]

File: apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.scss (hunk 29 above)
Comment (MadsBuchmann) [reply apps/cookbook/src/app/showcase/list-load-on-demand-showcase/list-load-on-demand-showcase.component.scss]: Actually stole it from `list-showcase.component.html` and `list-showcase.component.scss`. I will not fix those files use of `page-example` as part of this PR as the files are otherwise not related to this PR. 

Anyways should be resolved by: bed618b 😀

**Review by jakobe — APPROVED** [review]
🎉
